contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -6038,4 +6070,29 @@ remove_db_name_in_schema(const char *object_name, const char *object_type)
 	pfree(splited_object_name);
 
 	return (const char *)pstrdup(object_name + prefix_len);
-}
\ No newline at end of file
+}
+
+static int32
+tsql_get_numeric_typmod_from_exp(Plan *plan, Node *expr)
+{
+	int32       result_typmod = -1;
+	uint8_t     scale,
+                precision;
+	if (pltsql_protocol_plugin_ptr && *pltsql_protocol_plugin_ptr && (*pltsql_protocol_plugin_ptr)->get_numeric_typmod_from_exp)
+		result_typmod = (*pltsql_protocol_plugin_ptr)->get_numeric_typmod_from_exp(plan, expr);
+	if (result_typmod != -1)
+	{
+		/* 
+		* If we are unable to get correct precision and scale for overflow cases
+		* then return -1
+		*/
+		scale = (result_typmod - VARHDRSZ) & 0xffff;
+		precision = ((result_typmod - VARHDRSZ) >> 16) & 0xffff;
+		if (precision > TDS_NUMERIC_MAX_PRECISION)
+		{
+			if (!(precision - scale > 32 && scale > 6) && !(precision - scale <= TDS_NUMERIC_MAX_PRECISION))
+				return -1;
```

Comment:
Can you explain this?

Reply:
In function `resolve_numeric_typmod_from_exp()`, arithmetic operations such as division and multiplication might results in precision overflow cases, we had computed precision and scale for some cases but for other cases we keep the precision as it is. For such unhandled overflow cases the result typmod should be `-1`.